### ISSUE TITLE
"Download the full relation" error message not actionable

### ISSUE THREAD

**Issue by SomeoneElseOSM** [openstreetmap/iD#10316]
### URL

https://www.openstreetmap.org/edit?node=5290940#map=19/51.78760/-1.28748

### How to reproduce the issue?

Try and split way (which is part of a large relation)
The little icon that would allow you to split the way is greyed out.  Some transient text flashes on and off the screen saying "this line cannot be split".  It also says something like "Download the full relation".  No indication how to do that is provided.

### Screenshot(s) or anything else?

![Screenshot_20240711_094127](https://github.com/openstreetmap/iD/assets/5311740/86d3362c-d2c6-4e73-898b-a2e629fc0b87)

### Which deployed environments do you see the issue in?

Released version at openstreetmap.org/edit

### What version numbers does this issue effect?

2.29.0

### Which browsers are you seeing this problem on?

Firefox

**Comment by 1ec5** [comment]
This would be a little more intuitive if the message would name the relation that you’re supposed to download, and if we’d add a button for downloading it directly to the relation’s inspector rather than the inspector for each of its members: #5420.

**Comment by SomeoneElseOSM** [comment]
Changing the message to say "this is part of a relation - click here to download" would fix the problem (even if, as here, it would appear more than once).

I wasn't aware that that there was a button to "download a full relation" until I looked at the issues here - there's nothing in the relation UI (which I've looked at a lot!) to indicate that such a thing is possible or necessary.

**Comment by 1ec5** [comment]
Right, the existing button isn’t in an intuitive place, and it also takes up some needed screen real estate. I’m hoping the design for #5420 would address that to some extent.

The places where iD shows this message are transient and noninteractive. The bottom status banner would need to be replaced by the same kind of banner that appears when you lose your connection to the OSM API. (That banner has a Retry button on one end.)

**Comment by k-yle** [comment]
I think we could solve this issue by refactoring iD to support async actions. This means there'll never be an error message.

If you try to split a way, instead of erroring, we can:
1. show a spinner
2. download the relation which caused the error
3. then perform the original split operation

**Comment by jjiglesiasg** [comment]
To much talk and messages BUT NOT SOLUTION, in ID you CAN'T download very large relations whatever you want or not.

How you are suppose to split a way to include a bridge of something else, if you need to download the WHOLE relation of such way. 

THAT DO NOT MAKE SENSE AT ALL. Unnecessary restriction and limitation for ID users. Who invented this? a JOSM user? Another reason to walk away from ID and use RAPID...

**Comment by jjiglesiasg** [comment]
> I think we could solve this issue by refactoring iD to support async actions. This means there'll never be an error message.
> 
> If you try to split a way, instead of erroring, we can:
> 
> 1. show a spinner
> 2. download the relation which caused the error
> 3. then perform the original split operation

#Why splitting a road, requires to download a whole relation? You are not Deleting the road from the relation, just creating a new member in such relation. This limitation DO NOT MAKE SENSE AT ALL, and place an unneserary burden in ID users,,,

**Comment by 1ec5** [comment]
> Why splitting a road, requires to download a whole relation?

In some route relations, if you split a way without first downloading the full relation, iD (or Rapid) doesn’t have enough context to place the new member in the correct position(s) in the route relation’s list of members. #10149 introduced this requirement as part of a solution to many complaints about mangling relation member sort order.

**Comment by matkoniecz** [comment]
I am unable to reproduce it anymore.

**Comment by dschweisguth** [comment]
> I am unable to reproduce it anymore.

This is still an issue. For anyone else who finds this by searching for what to do when you get the message: click the download button(s) on the relation(s) which contain(s) the way you want to split, at the bottom of the edit pane.

**Comment by k-yle** [comment]
> I am unable to reproduce it anymore.

Yeah, this is still an extremely annoying issue, especially around coastlines. An example of how to reproduce it is in PR: https://github.com/openstreetmap/iD/pull/10615#issuecomment-4030150386
